File: java/sort.java
import java.util.*;

public class sort {

    public static void main(String[] args)
    {
        ArrayList<Integer> a = new ArrayList<>();

        a.add(3);
        a.add(7);
        a.add(1);
        a.add(4);
        a.add(10);
        a.add(9);
        a.add(20);
        a.add(12);
        a.add(13);
        a.add(5);
        a.add(77);
        a.add(7);

        System.out.println(a);

        a = insertion(a);

        System.out.println(a);

    }

    public static ArrayList<Integer> bubble(ArrayList<Integer> input_arr)
    {
        int temp;
        for(int i=input_arr.size()-1; i>0; i--)
        {
            for(int j=0; j<i; j++)
            {
                if(input_arr.get(j) > input_arr.get(j+1))
                {
                    temp = input_arr.get(j);
                    input_arr.set(j, input_arr.get(j+1));
                    input_arr.set(j+1, temp);
                }
            }
        }
        return input_arr;
    }

    public static ArrayList<Integer> selection(ArrayList<Integer> input_arr)
    {
        int temp;
        int idx = 0;
        for(int i=0; i<input_arr.size()-1; i++)
        {
            idx = i;
            for(int j=i+1; j<input_arr.size(); j++)
            {
                if(input_arr.get(idx) > input_arr.get(j))
                    idx = j;
            }
            temp = input_arr.get(i);
            input_arr.set(i, input_arr.get(idx));
            input_arr.set(idx, temp);
        }
        return input_arr;
    }


    public static ArrayList<Integer> insertion(ArrayList<Integer> input_arr)
    {
        int temp;
        int idx=0;
        for(int i=1; i<input_arr.size(); i++)
        {
            idx = i;
            temp = input_arr.get(i);
            for(int j=i-1; j>=0; j--)
            {  
                if(temp < input_arr.get(j))
                {
                    input_arr.set(idx, input_arr.get(j));
                    idx = j;
                    if(j==0)
                        input_arr.set(0, temp);
                }
                else
                {
                    input_arr.set(idx, temp);
                    break;
                }
            }
        }
        return input_arr;
    }

}

File: python/sort.py
class sort(object):
    def __init__(self):
        pass

    def bubble(self, sort_list):
        for i in range(len(sort_list)-1, 0, -1):
            for j in range(0, i):
                if sort_list[j] > sort_list[j+1]:
                    sort_list[j], sort_list[j+1] = sort_list[j+1], sort_list[j]
        return sort_list

    def selection(self, sort_list):
        for i in range(0, len(sort_list)):
            idx = i
            for j in range(i+1, len(sort_list)):
                if sort_list[idx] > sort_list[j]: idx=j
            sort_list[i], sort_list[idx] = sort_list[idx], sort_list[i]
            
        return sort_list

    def insertion(self, sort_list):
        for i in range(1, len(sort_list)):
            idx = i

            for j in range(i-1, -1, -1):
                if sort_list[idx] < sort_list[j]:
                    sort_list[idx], sort_list[j] = sort_list[j], sort_list[idx]
                    idx = j
                else:
                    break
        return sort_list

    def quick_partition(self, sort_list, left, right):
        if len(sort_list) == 1:
            return sort_list
        
        pivot = sort_list[left]
        print(pivot)

        while (left < right):
            if sort_list[left] < pivot:
                left += 1
                continue

            if sort_list[right] > pivot:
                right -= 1
                continue
            
            sort_list[left], sort_list[right] = sort_list[right], sort_list[left]

        sort_list[0], sort_list[left-1] = sort_list[left-1], sort_list[0]
        pivot_index = left

        return sort_list, pivot_index
    
    def quick(self, sort_list, left, right):
        if left < right:
            sort_list, pivot_index = self.quick_partition(sort_list, left, right)

            self.quick(sort_list, 0, pivot_index-1)
            self.quick(sort_list, pivot_index+1, right)

        return sort_list